design/KR/kenward.Rmd

```diff
@@ -0,0 +1,64 @@
+---
+title: how to implement Kenward Roger degree of freedom?
+author: "Liming Li"
+output: html_document
+editor_options:
+  chunk_output_type: console
+---
+
+# Objective
+
+The objective is to design how we can implement Kenward-Roger degree of freedom.
+
+# Current design of mmrm package
+
+In current mmrm package, we provided Satterthwaite degree of freedom, through `df_1d` and `df_md`.
+
+# Design of Kenward-Roger degree of freedom
+
+## update to current functions
+
+* `df_1d` and `df_md` renamed to `df_1d_sat` and `df_md_sat`, do not export them
+* add a new function `df_1d` and `df_md` to include degree of freedom selection
+* implement `df_1d_kr(2)` and `df_md_kr(2)` (suffix 2 means that it is improved Kenward-Roger degree of freedom)
```

Comment:
```suggestion
* implement `df_1d_kr2` and `df_md_kr2` (suffix 2 means that it is improved Kenward-Roger degree of freedom)
```
right because we don't want parentheses in the function names?

Reply:
yes we don't want parentheses, but here I mean `df_1d_kr` and `df_1d_kr2` for original and improved KR df

Reply:
Ah ok both versions. Ok or could do via an argument